src/lib.rs

```diff
@@ -59,20 +58,21 @@ pub struct Sk8brdMsg {
 }
 pub const MSG_HDR_SIZE: usize = size_of::<Sk8brdMsg>();
 
-pub fn send_msg(write_sink: &mut impl Write, r#type: Sk8brdMsgs, size: usize, buf: &[u8]) {
-    let msg: Sk8brdMsg = Sk8brdMsg {
-        r#type: r#type as u8,
-        len: size as u16,
-    };
-
-    write_sink
-        .write_all(&bincode::serialize(&msg).unwrap())
-        .unwrap();
-    let _ = write_sink.write(&buf[0..size]).unwrap();
+pub fn send_msg(
+    write_sink: &mut impl Write,
+    r#type: Sk8brdMsgs,
+    buf: &[u8],
+) -> Result<(), std::io::Error> {
+    let len = buf.len();
+    let hdr = [r#type as u8, (len & 0xff) as u8, ((len >> 8) & 0xff) as u8];
```

Comment:
Would it make more sense to turn this into two `write_all`s with `(len as u16).to_le_bytes()`, or is the writer not buffered?

Reply:
I think it's not buffered. Anyway, I'd wait for the `write_all_vectored` to stabilize and then use that one instead.